src/flb_input_thread.c

```diff
@@ -120,6 +120,10 @@ int flb_input_thread_destroy(struct flb_input_thread *it, struct flb_input_insta
 {
     int ret;
     flb_input_thread_exit(it, ins);
+    /* On Darwin, we must call pthread_cancel here to ensure worker
+     * thread termination. Otherwise, worker thread termination will
+     * be blocked. */
+    pthread_cancel(it->thread);
```

Comment:
FYI: On macOS, we must to call `pthread_cancel` here. Otherwise, termination operation of input thread will be blocked forever.